src/features/tokenGatedRooms/utils/isTokenGatedRoom.ts

```diff
@@ -0,0 +1,18 @@
+import { EnhancedStream } from '$/types'
+import getStream from '$/utils/getStream'
+import RoomNotFoundError from '$/errors/RoomNotFoundError'
+import { RoomId } from '$/features/room/types'
+import { StreamrClient } from 'streamr-client'
+
+export async function isTokenGatedRoom(roomId: RoomId, streamrClient: StreamrClient) {
```

Comment:
We use it in `pin.saga` and `getPrivacy` saga. In both cases we already have `roomId`-represented stream at hand.

Let's turn it into
```ts
export default function isTokenGatedRoom(stream: EnhancedStream) {
    return !!stream.extensions['thechat.eth'].tokenAddress
}
```